[BUG] Error in indexer when ordering make on ibetExchange

## Describe the bug

I get the following error when placing a make order on ibetExchange.

```
│   [2022-01-17 02:47:31 +0000] [INDEXER-POSITION-MEMBERSHIP] [37] [ERROR] Class 'builtins.NoneType' is not mapped                                                                                                                           │
│   Traceback (most recent call last):                                                                                                                                                                                                       │
│     File "batch/indexer_Position_Membership.py", line 331, in __sync_exchange                                                                                                                                                              │
│       self.sink.on_position(                                                                                                                                                                                                               │
│     File "batch/indexer_Position_Membership.py", line 67, in on_position                                                                                                                                                                   │
│       sink.on_position(*args, **kwargs)                                                                                                                                                                                                    │
│     File "batch/indexer_Position_Membership.py", line 115, in on_position                                                                                                                                                                  │
│       self.db.merge(position)                                                                                                                                                                                                              │
│     File "<string>", line 2, in merge                                                                                                                                                                                                      │
│     File "/home/apl/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2969, in merge                                                                                                                      │
│       object_mapper(instance)  # verify mapped                                                                                                                                                                                             │
│     File "/home/apl/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlalchemy/orm/base.py", line 296, in object_mapper                                                                                                                  │
│       return object_state(instance).mapper                                                                                                                                                                                                 │
│     File "/home/apl/.pyenv/versions/3.8.7/lib/python3.8/site-packages/sqlalchemy/orm/base.py", line 318, in object_state                                                                                                                   │
│       raise exc.UnmappedInstanceError(instance)                                                                                                                                                                                            │
│   sqlalchemy.orm.exc.UnmappedInstanceError: Class 'builtins.NoneType' is not mapped
```




Error conditions:
- The record of the position of the token to be ordered does not exist.
- A new make order is placed.

--- a/batch/indexer_Position_Bond.py
+++ b/batch/indexer_Position_Bond.py
@@ -106,6 +106,7 @@ class DBSink:
                 position.exchange_balance = exchange_balance
             if exchange_commitment is not None:
                 position.exchange_commitment = exchange_commitment
+            self.db.merge(position)
         elif any(value is not None and value > 0 for value in [balance, pending_transfer, exchange_balance, exchange_commitment]):
             LOG.debug(
                 f"Position created (Bond): token_address={token_address}, account_address={account_address}"
@@ -117,8 +118,7 @@ class DBSink:
             position.pending_transfer = pending_transfer or 0
             position.exchange_balance = exchange_balance or 0
             position.exchange_commitment = exchange_commitment or 0
-
-        self.db.merge(position)
+            self.db.add(position)
 
     def flush(self):
         self.db.commit()

--- a/batch/indexer_Position_Coupon.py
+++ b/batch/indexer_Position_Coupon.py
@@ -102,6 +102,7 @@ class DBSink:
                 position.exchange_balance = exchange_balance
             if exchange_commitment is not None:
                 position.exchange_commitment = exchange_commitment
+            self.db.merge(position)
         elif any(value is not None and value > 0 for value in [balance, exchange_balance, exchange_commitment]):
             LOG.debug(
                 f"Position created (Coupon): token_address={token_address}, account_address={account_address}"
@@ -112,8 +113,7 @@ class DBSink:
             position.balance = balance or 0
             position.exchange_balance = exchange_balance or 0
             position.exchange_commitment = exchange_commitment or 0
-
-        self.db.merge(position)
+            self.db.add(position)
 
     def flush(self):
         self.db.commit()

--- a/batch/indexer_Position_Membership.py
+++ b/batch/indexer_Position_Membership.py
@@ -101,6 +101,7 @@ class DBSink:
                 position.exchange_balance = exchange_balance
             if exchange_commitment is not None:
                 position.exchange_commitment = exchange_commitment
+            self.db.merge(position)
         elif any(value is not None and value > 0 for value in [balance, exchange_balance, exchange_commitment]):
             LOG.debug(
                 f"Position created (Membership): token_address={token_address}, account_address={account_address}"
@@ -111,8 +112,7 @@ class DBSink:
             position.balance = balance or 0
             position.exchange_balance = exchange_balance or 0
             position.exchange_commitment = exchange_commitment or 0
-
-        self.db.merge(position)
+            self.db.add(position)
 
     def flush(self):
         self.db.commit()

--- a/batch/indexer_Position_Share.py
+++ b/batch/indexer_Position_Share.py
@@ -119,8 +119,6 @@ class DBSink:
             position.exchange_commitment = exchange_commitment or 0
             self.db.add(position)
 
-        self.db.merge(position)
-
     def flush(self):
         self.db.commit()
 

--- a/batch/indexer_Position_Share.py
+++ b/batch/indexer_Position_Share.py
@@ -105,6 +105,7 @@ class DBSink:
                 position.exchange_balance = exchange_balance
             if exchange_commitment is not None:
                 position.exchange_commitment = exchange_commitment
+            self.db.merge(position)
         elif any(value is not None and value > 0 for value in [balance, pending_transfer, exchange_balance, exchange_commitment]):
             LOG.debug(
                 f"Position created (Share): token_address={token_address}, account_address={account_address}"
@@ -116,6 +117,7 @@ class DBSink:
             position.pending_transfer = pending_transfer or 0
             position.exchange_balance = exchange_balance or 0
             position.exchange_commitment = exchange_commitment or 0
+            self.db.add(position)
 
         self.db.merge(position)
 

